src/lib/core/cubeindex.tsx

```diff
@@ -0,0 +1,20 @@
+export interface CubeIndex {
+  readonly face: number;
+  readonly row: number;
+  readonly column: number;
+}
+
+export namespace CubeIndex {
```

Comment:
I think namespaces are generally discouraged (or rather, not preferred) in TS/JS. Will need @itstang to fact check me on this, but for es6 the file itself is effectively the "namespace":
https://michelenasti.com/2019/01/23/is-typescript-namespace-feature-deprecated.html
https://github.com/microsoft/TypeScript/issues/30994

Reply:
to keep the grouping you can also make it an object. Or, I see `dimension` as an argument for both functions. You could curry that out and make this a factory.

Reply:
+1 would probably recommend using es6 standard modules over namespaces

Reply:
Makes sense. I've moved the static factory methods into a factory object.